nanoFramework.Tools.DebugLibrary.Shared/PortTcpIp/PortTcpIpManager.cs

```diff
@@ -150,69 +164,91 @@ private void StopDeviceWatchersInternal()
 
 
         #region Methods to manage device list add, remove, etc
+        /// <summary>
+        /// Get the device that communicates via the network port, provided it has been added to the
+        /// list of known devices.
+        /// </summary>
+        /// <param name="networkDevice"></param>
+        /// <returns></returns>
+        public static NanoDeviceBase GetRegisteredDevice(NetworkDeviceInformation networkDevice)
+        {
+            if (networkDevice is not null)
+            {
+                var devices = NanoFrameworkDevices.Instance;
+                lock (devices)
```

Comment:
Same question here on the lock, is it ok to have potentially very long lock? Any possible side effect?

Reply:
No. All code in the locks concerns the manipulation of the list. All potentially long running code / interactions with external code is done outside the locks.

I've checked all list operations for potential problems. The original code was apparently created with the idea that there is exactly one PortSerialTcpIpManager, so if a device is a network device it must be managed by this instance of  PortSerialTcpIpManager. That is not necessarily the case, so I added some code to first verify that this instance is indeed managing the device. But all of that is done outside the locks.